src/Api/Controllers/OrganizationUsersController.cs

```diff
@@ -498,4 +506,36 @@ public async Task<ListResponseModel<OrganizationUserBulkResponseModel>> BulkRest
         return new ListResponseModel<OrganizationUserBulkResponseModel>(result.Select(r =>
             new OrganizationUserBulkResponseModel(r.Item1.Id, r.Item2)));
     }
+
+    private async Task HandleOrganizationUserUpdateAsync(OrganizationUser updatedOrganizationUser, Guid savingUserId,
+        OrganizationUserUpdateRequestModel model, int additionalSmSeatsRequired)
+    {
+        if (additionalSmSeatsRequired > 0)
+        {
+            var organization = await _organizationRepository.GetByIdAsync(updatedOrganizationUser.OrganizationId);
+            var update = new SecretsManagerSubscriptionUpdate(organization, true)
+                .AdjustSeats(additionalSmSeatsRequired);
+
+            await _updateSecretsManagerSubscriptionCommand.ValidateUpdate(update);
+
+            try
+            {
+                // Only autoscale (if required) after all validation has passed so that we know it's a valid request before
+                // updating Stripe
+                await _updateSecretsManagerSubscriptionCommand.UpdateSubscriptionAsync(update);
+                await _updateOrganizationUserCommand.UpdateUserAsync(updatedOrganizationUser, savingUserId,
+                    model.Collections?.Select(c => c.ToSelectionReadOnly()), model.Groups);
+            }
+            catch
+            {
+                // If we fail to update Stripe, we need to revert the update to Organization
+                await _organizationService.UpdateAsync(organization);
+            }
```

Comment:
This isn't quite right. The problem is that the sub update and the orgUser update need to pass together or fail together. So you can either do this:

1. validate subscription update
2. do the orgUser update (which includes validation)
4. do the subscription update
5. if (3) fails for some other reason - revert the orgUser update

OR this:

1. validate the orgUser update
2. do the subscription update (which includes validation)
3. do the orgUser update
4. if (3) fails for some other reason - revert the subscription update

In both cases, note that if (2) fails, there's nothing to revert yet.

Here, the `SecretsManagerSubscriptionUpdateCommand` already has a validate interface, and I think internal db changes are easier to revert than external Stripe charges, so I lean towards the first order of operations.

If you agree, compare it to your code - I think you need to swap the order you're doing the updates in (in the `try` block), and revert the orgUser updates, not the organization (in the `catch` block).